sublime_jedi/helper.py

```diff
@@ -33,50 +39,56 @@ def docstring_panel(view, docstring):
         sublime.status_message('Jedi: No results!')
 
 
-def docstring_tooltip(view, docstring, content_builder):
-    """Show docstring in popup.
-
-    :param view (sublime.View): current active view
-    :param docstring (basestring): python __doc__ string
-    :param content_builder (callable): callable object should accept docstring
-        and return content read for popup (a html text)
-    """
-    if docstring:
-        content = content_builder(docstring)
-        view.show_popup(content, max_width=512)
-    else:
-        sublime.status_message('Jedi: No results!')
-
-
-def simple_html_builder(docstring):
-    docstring = docstring.split('\n')
-    docstring[0] = '<b>' + docstring[0] + '</b>'
-    html = '<body><p style="font-family: sans-serif; font-family: sans-serif;">{0}</p></body>'.format(
-       '<br />'.join(docstring)
-    )
-    return html
-
-
 class SublimeJediDocstring(PythonCommandMixin, sublime_plugin.TextCommand):
     """Show docstring."""
 
     def run(self, edit):
         ask_daemon(self.view, self.render, 'docstring')
 
     def render(self, view, docstring):
+        if docstring is None or docstring == '':
+            logger.debug('Empty docstring.')
+            return
+
         if is_sublime_v2():
-            docstring_panel(view, docstring)
+            show_docstring_panel(view, docstring)
         else:
-            docstring_tooltip(view, docstring, simple_html_builder)
+            show_docstring_tooltip(view, docstring)
 
 
 class SublimeJediSignature(PythonCommandMixin, sublime_plugin.TextCommand):
-    """
-    Show signature in status bar
-    """
+    """Show signature in status bar."""
+
     def run(self, edit):
         ask_daemon(self.view, self.show_signature, 'signature')
 
     def show_signature(self, view, signature):
         if signature:
             sublime.status_message('Jedi: {0}'.format(signature))
+
+
+class SublimeJediTooltip(sublime_plugin.EventListener):
+    """EventListener to show jedi's docstring tooltip."""
+
+    def enabled(self):
+        """Check if hover popup is desired."""
+        return get_plugin_settings().get('enable_tooltip', True)
+
+    def on_activated(self, view):
+        """Handle view.on_activated event."""
+        if not (self.enabled() and view.match_selector(0, 'source.python')):
```

Comment:
I start thought, "how sublime colorize source code? is there a more reach scopes?" and have found next

https://www.sublimetext.com/docs/3/scope_naming.html#entity
https://www.sublimetext.com/docs/3/scope_naming.html#meta

for function it will be 
`source.python` `meta.function.python` `entity.name.function.python`

I think we can try to activate tooltip only for functions, classes and modules (packages), @deathaxe  what you think about this ?